### ISSUE TITLE
undefined: openapi3.CircularReferenceCounter with kin-openapi@v0.126.0

### ISSUE THREAD

**Issue by LucasMaciuga** [oapi-codegen/oapi-codegen#1702]
Hi,

I get "undefined: openapi3.CircularReferenceCounter" messages with kin-openapi@v0.126.0 and oapi-codegen@latest
Workaround fix is to set the version back to 0.125.0: `go get -u ./... github.com/getkin/kin-openapi@v0.125.0`

Output while "go generate"
```
# github.com/oapi-codegen/oapi-codegen/v2/pkg/util
C:\xxx\go\pkg\mod\github.com\oapi-codegen\oapi-codegen\v2@v2.3.0\pkg\util\loader.go:24:45: undefined: openapi3.CircularReferenceCounter
C:\xxx\go\pkg\mod\github.com\oapi-codegen\oapi-codegen\v2@v2.3.0\pkg\util\loader.go:26:12: undefined: openapi3.CircularReferenceCounter
C:\xxx\go\pkg\mod\github.com\oapi-codegen\oapi-codegen\v2@v2.3.0\pkg\util\loader.go:33:12: undefined: openapi3.CircularReferenceCounter
api.v1.go:1: running "go": exit status 1
```

My api.v1.go file:
```go
//go:generate go run github.com/oapi-codegen/oapi-codegen/v2/cmd/oapi-codegen --config=type.cfg.yaml api.yaml
//go:generate go run github.com/oapi-codegen/oapi-codegen/v2/cmd/oapi-codegen --config=client.cfg.yaml api.yaml
package api

....
```

type.cfg.yaml
```yaml
---
package: api
generate:
  models: true
output: api.type.gen.go
```

client.cfg.yaml
```yaml
---
package: api
generate:
  client: true
output: api.client.gen.go
```

My api.yaml Spec:
```yaml
---
openapi: '3.0.1'
info:
  title: Example
  version: '0.1'
  description: 'Some description'
servers:
  - url: http://localhost:8080
paths:
  /postme:
    post:
      operationId: PostIt
      description: Post
      tags:
        - Postman
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/exampleVals'
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Apple'
        '401':
          $ref: '#/components/responses/Unauthorized'
        '404':
          $ref: '#/components/responses/NotFound'

components:
  schemas:
    exampleVals:
      type: object
      properties:
        val1:
          type: string
        val2:
          type: string
      example:
        val1: "val1"
        val2: "val2"
    Apple:
      type: object
      properties:
        color:
          type: string
          enum:
            - red
            - green
            - blue
      example:
        color: "red"
    Error:
      type: object
      properties:
        Code:
          type: string
        Message:
          type: string
      example:
        Code: "ERR7801"
        Message: "Could not do anything!"
      required:
        - Code
        - Message
  responses:
    NotFound:
      description: The specified resource was not found
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/Error'
    Unauthorized:
      description: Unauthorized
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/Error'

```

**Comment by guicholeo** [comment]
the new version of github.com/getkin/kin-openapi/openapi3` does not have the `CircularReferenceCounter` I had to manually downgrade `go get github.com/getkin/kin-openapi/openapi3@v0.125.0`

**Comment by jamietanna** [comment]
Thanks for the report - and sorry for all who are affected by this - this is a known issue, and https://github.com/oapi-codegen/oapi-codegen/discussions/1692 has more details on how to resolve it until the next release

**Comment by jamietanna** [comment]
Closed as part of the v2.4.0 release
